DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/EditDelegateController.cs

```diff
@@ -0,0 +1,112 @@
+    [FeatureGate(FeatureFlags.RefactoredTrackingSystem)]
+    [Authorize(Policy = CustomPolicies.UserCentreAdmin)]
+    [Route("TrackingSystem/Delegates/{delegateId:int}/Edit")]
+    [SetDlsSubApplication(nameof(DlsSubApplication.TrackingSystem))]
+    public class EditDelegateController : Controller
+    {
+        private readonly CentreCustomPromptHelper centreCustomPromptHelper;
+        private readonly IJobGroupsDataService jobGroupsDataService;
+        private readonly IUserService userService;
+
+        public EditDelegateController(
+            IUserService userService,
+            IJobGroupsDataService jobGroupsDataService,
+            CentreCustomPromptHelper customPromptHelper
+        )
+        {
+            this.userService = userService;
+            this.jobGroupsDataService = jobGroupsDataService;
+            centreCustomPromptHelper = customPromptHelper;
+        }
+
+        [HttpGet]
+        public IActionResult Index(int delegateId)
+        {
+            var centreId = User.GetCentreId();
+            var delegateUser = userService.GetUsersById(null, delegateId).delegateUser;
+
+            if (delegateUser == null || delegateUser.CentreId != centreId)
+            {
+                return NotFound();
+            }
+
+            var jobGroups = jobGroupsDataService.GetJobGroupsAlphabetical().ToList();
+
+            var customPrompts =
+                centreCustomPromptHelper.GetEditCustomFieldViewModelsForCentre(delegateUser, centreId);
+            var model = new EditDelegateViewModel(delegateUser, jobGroups, customPrompts);
+
+            return View(model);
+        }
+
+        [HttpPost]
+        public IActionResult Index(EditDelegateFormData formData, int delegateId)
+        {
+            var centreId = User.GetCentreId();
+            if (!formData.JobGroupId.HasValue)
+            {
+                ModelState.AddModelError(nameof(EditDetailsFormData.JobGroupId), "Select a job group");
+            }
+
+            centreCustomPromptHelper.ValidateCustomPrompts(formData, centreId, ModelState);
+
+            if (!userService.NewAliasIsValid(formData.AliasId, delegateId, centreId))
+            {
+                ModelState.AddModelError(
+                    nameof(EditDelegateFormData.AliasId),
+                    "A user with this alias ID is already registered at this centre"
+                );
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return ReturnToEditDetailsViewWithErrors(formData, delegateId, centreId);
+            }
+
+            if (!userService.NewEmailAddressIsValid(formData.Email!, null, delegateId, centreId))
+            {
+                ModelState.AddModelError(
+                    nameof(EditDetailsFormData.Email),
+                    "A user with this email address is already registered at this centre"
+                );
+                return ReturnToEditDetailsViewWithErrors(formData, delegateId, centreId);
+            }

```

Comment:
Why do we not do this before the previous !ModelState.IsValid check?

Reply:
See this comment chain https://github.com/TechnologyEnhancedLearning/DLSV2/pull/696#discussion_r738471840